account_invoice_production_lot/models/account_invoice.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2011 Domsense s.r.l. <http://www.domsense.com>
+# Copyright 2013 Lorenzo Battistini <lorenzo.battistini@agilebg.com>
+# Copyright 2017 Vicent Cubells <vicent.cubells@tecnativa.com>
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    prod_lot_ids = fields.Many2many(
+        comodel_name='stock.production.lot',
+        compute='_compute_prod_lots',
+        string="Production Lots",
+    )
+
+    lot_formatted_note = fields.Html(
+        string='Formatted Note',
+        compute='_compute_line_lots',
+    )
+
+    @api.multi
+    def _compute_prod_lots(self):
+        for line in self:
+            line.prod_lot_ids = line.mapped(
+                'move_line_ids.move_line_ids.lot_id'
+            )
+
+    @api.multi
+    def _compute_line_lots(self):
+        for line in self:
+            note = '<ul>'
+            lot_strings = []
+            for sml in line.mapped('move_line_ids.move_line_ids'):
+                if sml.lot_id:
+                    if sml.product_id.tracking == 'serial':
+                        lot_strings.append('<li>%s %s</li>' % (
+                            _('S/N'), sml.lot_id.name,
+                        ))
+                    else:
+                        lot_strings.append('<li>%s %s (%s)</li>' % (
+                            _('Lot'), sml.lot_id.name, sml.qty_done,
+                        ))
+            if lot_strings:
+                note += ' '.join(lot_strings)
+                note += '</ul>'
+                line.lot_formatted_note = note
```

Comment:
Why not group by lots...? I attach a picture. By this way we cosume a lot vertical space
And is posible do this directly in report?

![Selección_299](https://user-images.githubusercontent.com/7689807/60825343-c8b07d80-a1ab-11e9-92ec-7aa07d64e119.png)

Reply:
> Why not group by lots...? I attach a picture. By this way we cosume a lot vertical space
> And is posible do this directly in report?
> 
> ![Selección_299](https://user-images.githubusercontent.com/7689807/60825343-c8b07d80-a1ab-11e9-92ec-7aa07d64e119.png)

Lots should be grouped correctly now